src/vos/vos_tree.c

```diff
@@ -188,27 +148,12 @@ ktr_rec_msize(int alloc_overhead)
 static void
 ktr_hkey_gen(struct btr_instance *tins, d_iov_t *key_iov, void *hkey)
 {
-	struct ktr_hkey		*kkey  = (struct ktr_hkey *)hkey;
+	struct ktr_hkey	*kkey = (struct ktr_hkey *)hkey;
 
-	if (key_iov->iov_len <= KH_INLINE_MAX) {
-		kkey->kh_hash[0] = 0;
-		kkey->kh_hash[1] = 0;
-
-		/** Set the lowest bit for inline key */
-		kkey->kh_inline_len = (key_iov->iov_len << 2) | 1;
-		memcpy(&kkey->kh_inline[0], key_iov->iov_buf, key_iov->iov_len);
-		D_ASSERT(kkey->kh_len & 1);
-		return;
-	}
+	hkey_common_gen(key_iov, hkey);
 
-	kkey->kh_murmur64 = d_hash_murmur64(key_iov->iov_buf, key_iov->iov_len,
-					    VOS_BTR_MUR_SEED);
-	kkey->kh_str32 = d_hash_string_u32(key_iov->iov_buf, key_iov->iov_len);
-	/** Lowest bit is clear for hashed key */
-	kkey->kh_len = key_iov->iov_len << 2;
-
-	vos_kh_set(kkey->kh_murmur64);
-	D_ASSERT(!(kkey->kh_inline_len & 1));
+	if (key_iov->iov_len > KH_INLINE_MAX)
```

Comment:
<=

Would it be better to return a boolean from hkey_common_gen, true if hashed?

Reply:
ah, nevermind, this is right. I missed the > vs <.  Comment is a minor suggestion